Project2015To2017/Analysis/ReporterBase.cs

```diff
@@ -0,0 +1,41 @@
+using System.IO;
+using Project2015To2017.Transforms;
+
+namespace Project2015To2017.Analysis
+{
+	public abstract class ReporterBase
```

Comment:
I would rather use an interface here, this class doesn’t really do anything. You can achieve the same thing with an interface which is more flexible

Reply:
C# doesn't have default methods in interface, does it? And I'm not sure about sth being allowed to be a Reporter and something else at the same time.

Reply:
I think inheritance in implementation code (not in objects) is used to easily over just plain interfaces. Base classes have a tendency to become big. 

I’m not that eager on this change though. They are both valid solutions to the same problem

Reply:
If you keep this class, at least make it implement an interface so you can use that in other classes as to not tie it to a concrete type. In order to implement the behavior it doesn’t have to derive from `ReporterBase`.